src/components/ssl-observatory.js

```diff
@@ -313,7 +313,12 @@ SSLObservatory.prototype = {
     function toHexString(charCode) {
       return ("0" + charCode.toString(16)).slice(-2);
     }
-    return [toHexString(h.charCodeAt(i)) for (i in h)].join("").toUpperCase();
+
+    var hexArr = [];
+    for (i in h){
+      hexArr.push(toHexString(h.charCodeAt(i)));
```

Comment:
Have you tested this loop? Suggestion: run before and after the patch, look at submitted hex values from the logs for a known host (e.g. libssh.org), and make sure they are the same.